filebeat/input/event.go

```diff
@@ -39,8 +39,19 @@ func (e *Event) ToMapStr() common.MapStr {
 		"input_type":            e.InputType,
 	}
 
-	if e.JSONConfig != nil && len(e.JSONFields) > 0 {
-		mergeJSONFields(e, event)
+	// Add data fields which are added by the readers
+	for key, value := range e.Data {
+		event[key] = value
+	}
+
+	// Check if json fields exist
+	var jsonFields common.MapStr
+	if fields, ok := event["json"]; ok {
+		jsonFields = fields.(common.MapStr)
+	}
```

Comment:
I want to have the Event to be generic and not specifically having JSON inside. Remember the origin of this PR is when adding data for the multiline reader. So instead of adding more and more variable to event, data is the generic container for it. 

Should we cleanup JSON? Yes, definitively. I even think the processing should probably not happen in the event package.